### ISSUE TITLE
Is more than 10% battery usage normal?

### ISSUE THREAD

**Issue by jayb-g** [binwiederhier/ntfy-android#90]
I tested this app on a degoogled android and it gives more than 10% battery usage. The least I have seen is 4-5%. 

This is on Mobile data, no WiFi.
On (mostly)WiFi it seems to work fine with <1% battery usage.

All default settings. With only molly unifiedpush setup with it.

Battery optimization is disabled for both ntfy and molly-up

Is this normal battery usage for ntfy?

Android 14 latest on  Pixel 7

**Comment by Kuchteq** [comment]
On Pixel 8 GrapheneOS it is around 50 for me, often more than that 
![Screenshot_20241002-215341](https://github.com/user-attachments/assets/043fbe24-2146-43d7-bc87-08765f6e496a)
Don't know what causes it

**Comment by etam** [comment]
I just noticed this as well:
![Screenshot_20241003-090556_Ustawienia](https://github.com/user-attachments/assets/4b9688c6-7b61-4e56-a948-d4616a678438)

**Comment by titulebolide** [comment]
Same issue, ntfy app is responsable of half of my battery usage. I'm using unified push.

**Comment by Glareascum** [comment]
I don't have this problem, I'm using unified push, and with 4hours of background work it's just <1% battery. I'm on LineageOS Android 14

**Comment by titulebolide** [comment]
@Glareascum this is interesting!
Did you disabled battery optimisation?
Are you using websockets?
I'm running e/OS 2.x which is based on Lineage (18 IIRC)

**Comment by Glareascum** [comment]
> @Glareascum this is interesting! Did you disabled battery optimisation? Are you using websockets? I'm running e/OS 2.x which is based on Lineage (18 IIRC)

Yep I disabled battery optimisation and using websockets. I'm on LineageOS 21

**Comment by Glareascum** [comment]
As per #1 , you should open a ticket on the main repo to get more support

**Comment by Glareascum** [comment]
> > I don't have this problem, I'm using unified push, and with 4hours of background work it's just <1% battery. I'm on LineageOS Android 14.
> > I'm using it for unified push and custom topics on a self hosted instance
> 
> Was this entirely/mostly on wifi ?

Yes mostly on wifi, but there are days where I'm mostly in lte and I didn't notice anything different

**Comment by titulebolide** [comment]
@Glareascum do you self host your ntfy server? If yes, would you mind giving a peak to the kind of perf you are having with your server? Mine is not the quickest fox in the forest, but I can hardly imagine its low specs being linked to my battery drainage.

**Comment by titulebolide** [comment]
Hi, I noticed yesterday that battery optimisation was enabled for ntfy. To match @Glareascum setup, I disabled it. My battery drainage significantly dropped and is below 1% now. At the same time yesterday it was over 10%.

**Comment by Ilav1** [comment]
I also have battery draining problem on Adndroid 14 (Pixel 6a).
When battery optimisation is disabled everything is working fine but the battery usage of ntfy is very high.
When battery optimisation is enabled, I dont get the notifications immediately. There are hours between sending and recieving on mobile phone...

**Comment by binwiederhier** [comment]
(Please use the [main issue tracker](https://github.com/binwiederhier/ntfy/issues) going forward.)
